doc/production-hosting.md

```diff
@@ -1,40 +1,231 @@
+## Attached state
 
-See https://github.com/badges/ServerScript for helper admin scripts.
+Shields has mercifully little persistent state:
+
+1.  The GitHub tokens we collect are saved on each server in JSON files on disk.
+    They can be fetched from the [GitHub auth admin endpoint][] for debugging.
+    @paulmelnikow and @espadrine have access. The GitHub OAuth application is
+    part of @espadrine's account but
+    [it should be transferred to the badges org][oauth transfer].
+2.  The analytics data is also saved on each server in JSON files on disk.
+3.  The server keeps a few caches in memory. These are neither persisted nor
+    inspectable.
+    - The [request cache][]
+    - The [regular-update cache][]
+    - The [raster cache][]
+
+[github auth admin endpoint]: https://github.com/badges/shields/blob/master/services/github/auth/admin.js
+[request cache]: https://github.com/badges/shields/blob/master/lib/request-handler.js#L29-L30
+[regular-update cache]: https://github.com/badges/shields/blob/master/lib/regular-update.js
+[raster cache]: https://github.com/badges/shields/blob/master/gh-badges/lib/svg-to-img.js#L9-L10
+[oauth transfer]: https://developer.github.com/apps/managing-oauth-apps/transferring-ownership-of-an-oauth-app/
+
+## Configuration
+
+To bootstrap the configuration process,
+[the script that starts the server][start-shields.sh] sets a single
+environment variable:
+
+```
+NODE_CONFIG_ENV=shields-io-production
+```
+
+With that variable set, the server ([using `config`][config]) reads these
+files:
+
+- [`local-shields-io-production.yml`][local-shields-io-production.yml].
+  This file contains secrets which are checked in with a deploy commit.
+- [`shields-io-production.yml`][shields-io-production.yml]. This file
+  contains non-secrets which are checked in to the main repo.
+- [`default.yml`][default.yml]`. This file contains defaults.
+
+[start-shields.sh]: https://github.com/badges/ServerScript/blob/master/start-shields.sh#L7
+[config]: https://github.com/lorenwest/node-config/wiki/Configuration-Files
+[local-shields-io-production.yml]: ../config/local-shields-io-production.example.yml
+[shields-io-production.yml]: ../config/shields-io-production.yml
+[default.yml]: ../config/default.yml
+
+The project ships with `dotenv`, however there is no `.env` in production.
+
+## Badge CDN
+
+Sitting in front of the three servers is a Cloudflare Free account which
+provides several services:
+
+- Global CDN, caching, and SSL gateway for `img.shields.io`
+- Analytics through the Cloudflare dashboard
+- DNS hosting for `shields.io`
+
+Cloudflare is configured to respect the servers' cache headers.
+
+## Frontend
+
+The frontend is served by [GitHub Pages][] via the [gh-pages branch][gh-pages]. SSL is enforced.
+
+`shields.io` resolves to the GitHub Pages hosts. It is not proxied through
+Cloudflare.
+
+Any maintainer can push to `gh-pages` though in practice this needs to be kept
+in sync with the production deploy.
```

Comment:
are there any rollback related steps with the frontend, especially as it relates to keeping in sync with a server/backend rollback?

Reply:
A rollback is the same as a regular deploy, so the process is exactly the same.

The most likely way a rollback happens is after a bad canary deploy.

Also worth mentioning: you can, though normally should not, deploy commits made to a local branch.

Want to update where needed to clarify that?

Reply:
> though in practice this needs to be kept
in sync with the production deploy

This is the part I was thinking through. My first time reading through from top to bottom I didn't have context on this yet. What if we swapped the order of the `Deployment` and the `Frontend` sections? 

Although I suppose it doesn't really matter, this doc has a very specific, target audience (not the masses) 😄

Reply:
Another possibility is to consolidate all the "who has access to what" into a table at the end. We could qualify the point there. The reason to mention it is that while _technically_ anyone with push access to the repo can deploy the frontend, in practice they can't.

Reply:
I'm def 👍 💯 on that!

Reply:
Do you think it makes sense to keep the order of sections where it is? It's tricky because you can't explain the deploy without explaining the infrastructure.

Reply:
Yeah the order makes sense now. I like the verbiage you added in the table too. What if we reused the sentiment of the text you added in the table down in the `Frontend` section? 

in the Frontend section, replace:
 `Any maintainer can push to gh-pages though in practice this needs to be kept in sync with the production deploy` 
With:
`Technically any maintainer can push to gh-pages, but in practice this must be deployed with the badge server via the deployment process described below`